src/overlays/actors/ovl_Bg_Spot18_Basket/z_bg_spot18_basket.c

```diff
@@ -21,39 +36,397 @@ const ActorInit Bg_Spot18_Basket_InitVars = {
+        tempVector.y = actor->posRot.pos.y + 170.0f;
+        tempVector.z = actor->posRot.pos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7D50.s")
+        if (this->unk_218 == 0) {
+            for (i = 0; i < ARRAY_COUNT(D_808B85E4); i++) {
+                collectible = Item_DropCollectible(globalCtx, &tempVector, 4);
+                if (collectible != NULL) {
+                    collectible->actor.velocity.y = 11.0f;
+                    collectible->actor.posRot.rot.y = D_808B85E4[i];
+                }
+            }
+        } else if (this->unk_218 == 1) {
+            for (i = 0; i < ARRAY_COUNT(D_808B85E4); i++) {
+                collectible = Item_DropCollectible(globalCtx, &tempVector, 0);
+                if (collectible != NULL) {
+                    collectible->actor.velocity.y = 11.0f;
+                    collectible->actor.posRot.rot.y = D_808B85E4[i];
+                }
+            }
+        } else if (this->unk_218 == 2) {
+            if ((this->unk_21A != 0) || (Flags_GetCollectible(globalCtx, (actor->params & 0x3F)) != 0)) {
+                collectible = Item_DropCollectible(globalCtx, &tempVector, 20);
+                if (collectible != NULL) {
+                    collectible->actor.velocity.y = 11.0f;
+                    collectible->actor.posRot.rot.y = D_808B85E4[1];
+                }
+            } else {
+                collectible = Item_DropCollectible(globalCtx, &tempVector, ((actor->params & 0x3F) << 8) | 6);
+                if (collectible != NULL) {
+                    collectible->actor.velocity.y = 11.0f;
+                    collectible->actor.posRot.rot.y = D_808B85E4[1];
+                    this->unk_21A = 1;
+                }
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7F74.s")
+            collectible = Item_DropCollectible(globalCtx, &tempVector, 2);
+            if (collectible != NULL) {
+                collectible->actor.velocity.y = 11.0f;
+                collectible->actor.posRot.rot.y = D_808B85E4[0];
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7FC0.s")
+            collectible = Item_DropCollectible(globalCtx, &tempVector, 1);
+            if (collectible != NULL) {
+                collectible->actor.velocity.y = 11.0f;
+                collectible->actor.posRot.rot.y = D_808B85E4[2];
+            }
+        }
+    } else if (this->unk_216 == 2) {
+        if (this->unk_218 == 2) {
+            func_80078884(NA_SE_SY_CORRECT_CHIME);
+        } else {
+            func_80078884(NA_SE_SY_TRE_BOX_APPEAR);
+        }
+    } else if (this->unk_216 == 200) {
+        func_808B7BB0(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B818C.s")
+void BgSpot18Basket_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot18Basket* this = THIS;
+    Vec3s somePointer;
```

Comment:
```suggestion
    UNK_TYPE temp;
```

Reply:
I renamed it but both UNK_TYPE and UNK_PTR don't appear to match.